### PR TITLE
[New model] Support HY-Image3.0 DiT

### PR DESCRIPTION
**Pull request by skf-1999**
PLEASE FILL IN THE PR DESCRIPTION HERE ENSURING ALL CHECKLIST ITEMS (AT THE BOTTOM) HAVE BEEN CONSIDERED.

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft.
</details>

**BEFORE SUBMITTING, PLEASE READ <https://github.com/vllm-project/vllm-omni/blob/main/CONTRIBUTING.md>** (anything written below this line will be removed by GitHub Actions)

### CODE REVIEW

**Comment by hsliuustc0106** [comment]
a few comments and todo list:

- [ ] remove NPU related files
- [ ] remove EP support, focuse on TP now
- [ ] add test plan and results for t2i jobs
- [ ] update docs with model supports

**Comment by hsliuustc0106** [comment]
fix conflicts and remove NPU worker changes, submit another PR for NPU adaptation after #774 merged

**Comment by lishunyang12** [comment]
I will review this PR

**Review by lishunyang12 — COMMENTED** [review]
Large PR with significant debug artifacts, missing imports, hardcoded device refs.

File: vllm_omni/config/model.py [hunk 1]
```diff
@@ -115,6 +115,7 @@ def __post_init__(
         skip_mm_profiling: bool | None,
         video_pruning_rate: float | None,
     ) -> None:
+        print("-------OmniModelConfig--------")
```
Comment (lishunyang12) [inline vllm_omni/config/model.py:118]: Looks like a debug `print` statement was left in here — mind removing it?

File: vllm_omni/diffusion/layers/fused_moe.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
+import torch
+
+# TODO(bnell): Add shared + fused combo function? e.g. +
+class SharedFusedMoE(FusedMoE):
```
Comment (lishunyang12) [inline vllm_omni/diffusion/layers/fused_moe.py:4]: `FusedMoE`, `get_tensor_model_parallel_world_size`, and `tensor_model_parallel_all_reduce` are used but I do not see corresponding imports — this looks like it would raise a `NameError` at import time. Also, minor nit: missing a newline at end of file.

File: vllm_omni/diffusion/worker/gpu_worker.py [hunk 3]
```diff
@@ -136,7 +136,12 @@ def execute_model(self, reqs: list[OmniDiffusionRequest], od_config: OmniDiffusi
         # Refresh cache context if needed
         if self.cache_backend is not None and self.cache_backend.is_enabled():
             self.cache_backend.refresh(self.pipeline, req.num_inference_steps)
-        with set_forward_context(vllm_config=self.vllm_config, omni_diffusion_config=self.od_config):
+
+        from vllm_ascend.ascend_forward_context import MoECommType, set_ascend_forward_context
```
Comment (lishunyang12) [inline vllm_omni/diffusion/worker/gpu_worker.py:140]: This unconditionally imports `vllm_ascend`, which is an NPU-specific package — I think this would cause an `ImportError` in GPU-only environments that do not have it installed. Would it make sense to guard this behind an `is_npu()` check?

File: vllm_omni/diffusion/models/hunyuan/hunyuan_image3_utils.py [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+        cached_key = [key[:cached_prompt_len], key[seq_len - 1 : seq_len]]
+        cached_value = [value[:cached_prompt_len], value[seq_len - 1 : seq_len]]
+
+        if bs > 1:
+            assert bs == 2, "for cfg case, bs must be 2"
+            cached_key.append(key[seq_len : seq_len + cached_prompt_len])
+            cached_key.append(key[-1:])
+
+            cached_value.append(value[seq_len : seq_len + cached_prompt_len])
+            cached_value.append(value[-1:])
```
Comment (lishunyang12) [inline vllm_omni/diffusion/models/hunyuan/hunyuan_image3_utils.py:227]: `get_full_state_dict` loads every safetensors shard into CPU memory at once via `glob` + `load_file`. For large models this will OOM. Also, `safetensors`, `os`, and `glob` imports are inlined mid-file rather than at the top -- is this intentional?

File: vllm_omni/diffusion/models/hunyuan/hunyuan_image_3.py [hunk 5]
```diff
@@ -0,0 +1,953 @@
...
+            "final_layer",
+        ]
+        skip_prefixes.extend(unexpected_keywords)
+        loader = AutoWeightsLoader(
+            self,
+            skip_prefixes=skip_prefixes,
+        )
+        return loader.load_weights(weights)
+
+    @property
```
Comment (lishunyang12) [inline vllm_omni/diffusion/models/hunyuan/hunyuan_image_3.py:199]: `pre_load` calls `get_full_state_dict` which loads all safetensors shards into memory, then does `load_state_dict(..., strict=False)` and silently ignores `missing`/`unexpected`. This duplicates work that `AutoWeightsLoader` in `load_weights` already handles. Could this whole method and the double-load be avoided?

File: vllm_omni/diffusion/models/hunyuan/hunyuan_image_3_models.py [hunk 6]
```diff
@@ -0,0 +1,3541 @@
...
+
+        num_attention_heads = self.config.num_attention_heads
+        num_kv_heads = getattr(
+            self.config, "num_key_value_heads", self.config.num_attention_heads
+        )
+        split_params_mapping = [
+            (".gate_up_proj", ".gate_and_up_proj", 2, [(1, 1), (0, 1)], None),
+            (
+                ".qkv_proj",
+                ".qkv_proj",
```
Comment (lishunyang12) [inline vllm_omni/diffusion/models/hunyuan/hunyuan_image_3_models.py:1750]: I noticed quite a few commented-out `save_tensor` / `torch.load` / debug-path lines scattered throughout the diff. Would it make sense to clean these up before merge? They add noise and make the diff harder to review.

File: vllm_omni/diffusion/models/hunyuan/hunyuan_image_3_models.py [hunk 7]
```diff
@@ -0,0 +1,3541 @@
...
+                if weight_name == ".q_proj":
+                    match = re.search(r"layers\.\d+", name)
+                    if match:
+                        layer_id = int(match.group(0).split(".")[-1])
+                        if cla_factor > 1 and layer_id % cla_factor != 0:
+                            continue
+                name = name.replace(weight_name, param_name)
+                # Skip loading extra bias for GPTQ models.
+                if name.endswith(".bias") and name not in params_dict:
+                    continue
```
Comment (lishunyang12) [inline vllm_omni/diffusion/models/hunyuan/hunyuan_image_3_models.py:1827]: `save_weight_heatmap` imports `matplotlib` and `seaborn` at module scope, which adds fairly heavy visualization dependencies. This looks like it might be debugging/dev-only code — was there a reason to keep it in, or would it make sense to remove it before merge?

File: vllm_omni/diffusion/models/hunyuan/hunyuan_image_3_models.py [hunk 8]
```diff
@@ -0,0 +1,3541 @@
...
+                    # this is an expert weight and should not be
+                    # attempted to load as other weights later
+                    is_expert_weight = True
+
+                    # Do not modify `name` since the loop may continue here
+                    # Instead, create a new variable
+                    name_mapped = name.replace(weight_name, param_name)
+                    found_num += 1
+                    # if tp_rank == 0:
+                    #     print(f"name_mapped: {name_mapped}, found_num: {found_num}")
```
Comment (lishunyang12) [inline vllm_omni/diffusion/models/hunyuan/hunyuan_image_3_models.py:1913]: The norm initialization logic seems to unconditionally overwrite both with vLLM's \ — if that's the case, the \ dispatch above might never actually take effect. Is this intentional, or should one of them be removed?

File: vllm_omni/entrypoints/omni.py [hunk 9]
```diff
@@ -289,7 +289,7 @@ def _process_stage_ready(self, stage: OmniStage, stage_id: int, result: dict[str
         self._stages_ready.add(stage_id)
         logger.info(f"[{self._name}] Stage-{stage_id} reported ready")
 
-    def _wait_for_stages_ready(self, timeout: int = 120) -> None:
+    def _wait_for_stages_ready(self, timeout: int = 3000) -> None:
```
Comment (lishunyang12) [inline vllm_omni/entrypoints/omni.py:292]: Timeout bumped from 120s to 3000s (50 minutes). I wonder if this might be masking a slow model-loading path rather than addressing the root cause. Would it make sense to make this configurable instead of hardcoded?

File: examples/offline_inference/image_to_image/image_edit.py [hunk 10]
```diff
@@ -273,7 +279,12 @@ def main():
     # Enable VAE memory optimizations on NPU
     vae_use_slicing = is_npu()
     vae_use_tiling = is_npu()
-    parallel_config = DiffusionParallelConfig(ulysses_degree=args.ulysses_degree, ring_degree=args.ring_degree)
+    parallel_config = DiffusionParallelConfig(
+        tensor_parallel_size=args.tensor_parallel_size,
+        ulysses_degree=args.ulysses_degree,
```
Comment (lishunyang12) [inline examples/offline_inference/image_to_image/image_edit.py:284]: `enable_expert_parallel=True` is hardcoded here. Not every model uses MoE -- should this default to `False` or be driven by a CLI arg?
